mne/preprocessing/maxwell.py

```diff
@@ -0,0 +1,530 @@
+
+    return np.c_[rad, az, pol]
+
+
+# TODO: Eventually refactor this in forward computation code
+
+def _make_coils(info, accurate=True, elekta_defs=False):
+    """Prepare dict of MEG coils and their information.
+
+    Parameters
+    ----------
+    info : instance of mne.io.meas_info.Info | str
+        If str, then it should be a filename to a Raw, Epochs, or Evoked
+        file with measurement information. If dict, should be an info
+        dict (such as one from Raw, Epochs, or Evoked).
+    accurate : bool
+        Accuracy of coil information.
+    coil_def : str | None
+        Filepath to the coil definitions file.
+
+    Returns
+    -------
+    megcoils, meg_info : dict
+        MEG coils and information dict
+    """
+
+    if accurate:
+        accuracy = FIFF.FWD_COIL_ACCURACY_ACCURATE
+    else:
+        accuracy = FIFF.FWD_COIL_ACCURACY_NORMAL
+    meg_info = None
+    megnames, megcoils, compcoils = [], [], []
+
+    # MEG channels
+    picks = pick_types(info, meg=True, eeg=False, ref_meg=False,
+                       exclude=[])
+    nmeg = len(picks)
+    if nmeg > 0:
+        megchs = pick_info(info, picks)['chs']
+
+    if nmeg <= 0:
+        raise RuntimeError('Could not find any MEG channels')
+
+    meg_info = pick_info(info, picks) if nmeg > 0 else None
+
+    # Create coil descriptions with transformation to head or MRI frame
+    if elekta_defs:
+        elekta_coil_defs = op.join(op.split(__file__)[0], '..', 'data',
+                                   'coil_def_Elekta.dat')
+        templates = _read_coil_defs(elekta_coil_defs)
+
+        # Check that we have all coils needed
+        template_set = set([coil['coil_type'] for coil in templates['coils']])
+        req_coil_set = set([coil['coil_type'] for coil in meg_info['chs']])
+        if not req_coil_set.issubset(template_set):
+            warnings.warn('Didn\'t locate find enough Elekta coil definitions,'
+                          ' using default MNE coils.')
+            templates = _read_coil_defs()
+    else:
+        templates = _read_coil_defs()
+
+    if nmeg > 0:
+        # TODO: In fwd solution code, reformulate check that forces head
+        # coords and remove this hack. (Or use only head coords)
+        #info['dev_head_t']['trans'] = np.eye(4)  # Uncomment for device coords
+        megcoils = _create_coils(megchs, accuracy, info['dev_head_t'], 'meg',
+                                 templates)
+
+    return megcoils, meg_info
+
+
+def _update_info(raw, origin, int_order, ext_order, nsens, nmoments):
+    """Helper function to update info after Maxwell filtering."""
+
+    info_dict = dict(int_order=int_order, ext_order=ext_order,
+                     origin=origin, nsens=nsens, nmoments=nmoments,
+                     creator='MNE\'s Maxwell Filter')
+    raw.info['maxshield'] = False
+
+    # Insert information in raw.info['proc_info']
+    if 'proc_history' in raw.info.keys():
+        raw.info['proc_history'].insert(0, info_dict)
```

Comment:
That seems to be how MaxFilter is doing it. I think they're making a stack or something?